app/routes/authenticate.js

```diff
@@ -0,0 +1,20 @@
+const azureAuth = require('../azure-auth')
+
+module.exports = {
+  method: 'GET',
+  path: '/authenticate',
+  options: {
+    auth: { mode: 'try' }
+  },
+  handler: async (request, h) => {
+    try {
+      await azureAuth.authenticate(request.query.code, request.cookieAuth)
+      return h.redirect('/')
+    } catch (err) {
+      console.log('Error authenticating')
+      console.log(JSON.stringify(err))
+    }
+
+    return h.view('500').code(500)
```

Comment:
Would this be an error in the authentication process and there 500 is fine.  Or would failed authentication throw an error in which case 401 or 403 would be a better fit?

Reply:
This is for an error in the auth process so think code 500 is correct